docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc

```diff
@@ -195,3 +195,8 @@ Up to this version, the resolving of fallback messages was inconsistent across t
 The implementation has now been unified for all themes. In general, the message for the most specific matching language tag has the highest priority. If there are both a realm localization message and a Theme 18n message, the realm localization message has the higher priority. Summarized, the priority of the messages is as follows  (RL = realm localization, T = Theme i18n files): `RL <variant> > T <variant> > RL <region> > T <region> > RL <language> > T <language> > RL en > T en`.
 
 Probably this can be better explained with an example: When the variant `de-CH-1996` is requested and there is a realm localization message for the variant, this message will be used. If such a realm localization message does not exist, the Theme i18n files are searched for a corresponding message for that variant. If such a message does not exist, a realm localization message for the region (`de-CH`) will be searched. If such a realm localization message does not exist, the Theme i18n files are searched for a message for that region. If still no message is found, a realm localization message for the language (`de`) will be searched. If there is no matching realm localization message, the Theme i18n files are be searched for a message for that language. As last fallback, the English (`en`) translation is used: First, an English realm localization will be searched - if not found, the Theme 18n files are searched for an English message.
+
+= LDAPStorageProvider search changes
+
+Starting with this release Keycloak uses a pagination mechanism when querying federated LDAP database. 
+Searching for users should be consistent with search in local database.
```

Comment:
I would prefer to have also something about performance (unless I am wrong with something in my comment herehttps://github.com/keycloak/keycloak/issues/20156#issuecomment-1553307913 ). Or eventually about the behaviour change (if we go with the path of hardcoded "count" methods in LDAPStorageProvider as proposed by @mhajas).

Reply:
I'll address it within follow-up https://github.com/keycloak/keycloak/pull/20447

Reply:
Follow-up is fine. Thanks! I am then resolving this conversation and approving this PR